editoast/editoast_schemas/src/rolling_stock/train_category.rs

```diff
@@ -0,0 +1,28 @@
+use serde::Deserialize;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+use crate::rolling_stock::TrainMainCategory;
+
+editoast_common::schemas! {
+    TrainCategory,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ToSchema)]
+#[serde(untagged)]
+pub enum TrainCategory {
+    Main { main_category: TrainMainCategory },
+    Sub { sub_category_code: String },
```

Comment:
```suggestion
    Main(TrainMainCategory),
    Sub(String),
```

nit: prevents repetition

Reply:
This will change the api, TrainMainCategory is a string also

Reply:
you're right 🤦